src/modules/users/services/users.service.ts

```diff
@@ -11,4 +12,27 @@ export class UsersService {
   async findAll(queryUsersArgsDto: any) {
     return this.prismaService.user.findMany();
   }
+
+  // find user by phoneNumber and validate password
+  async findOneByPhoneNumber(phoneNumber: string) {
```

Comment:
`findOneByPhoneNumber` and `findOneById` is redundant, can be simplified into one function called `findOneOrThrow`